subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -526,7 +577,12 @@ static int z_ztest_run_test_suite_ptr(struct ztest_suite_node *suite)
 	TC_SUITE_START(suite->name);
 	test_result = ZTEST_RESULT_PENDING;
 	phase = TEST_PHASE_SETUP;
-	if (suite->setup != NULL) {
+#ifndef KERNEL
+	if (setjmp(test_suite_fail)) {
+		test_result = ZTEST_RESULT_SUITE_FAIL;
+	}
+#endif
```

Comment:
@asemjonovs I think I need `#else` here to handle the signals safely and not kill the main thread.